models/classes/runner/providers/ProviderRegistry.php

```diff
@@ -46,4 +46,37 @@ protected function getExtension()
     {
         return common_ext_ExtensionsManager::singleton()->getExtensionById('taoTests');
     }
+
+    /**
+     * Get all modules that belong to a category
+     * @param string $category - the provider category (runner, proxy,etc.)
+     * @return array the matching providers
+     */
+    public function getByCategory($category = null)
+    {
+        if (is_null($category)) {
+            return [];
+        }
+
+        return array_filter(
+            $this->getMap(),
+            function($provider) use ($category) {
+                return isset($provider['category']) && $provider['category'] == $category;
+            }
+        );
+    }
+
+    /**
+     * Unregister all modules that belong to a category
+     * @param string $category - the provider category (runner, proxy,etc.)
+     */
+    public function removeByCategory($category = null)
+    {
+        $providers = $this->getByCategory($category);
+        foreach ($providers as  $provider) {
```

Comment:
Please remove extra space after `as`

Reply:
`foreach ($this->getByCategory($category) as  $provider) {` would be better